### PR TITLE
#162414123 Connect profile components to redux state

### PR DESCRIPTION
**Pull request by eltNEG**
#### What does this PR do?
Connect profile components to redux state

#### Description of Task to be completed?
- Connect profile components to redux state

#### How should this be manually tested?
pull the repository and cd into the branch and then run `npm test`

#### Any background context you want to provide?
This is a continuation of #71 

#### What are the relevant pivotal tracker stories?
PT link - https://www.pivotaltracker.com/story/show/162414123

vup

### CODE REVIEW

File: src/containers/ProfileContainer/index.js [hunk 1]
```diff
@@ -1,27 +1,33 @@
...
     this.renderProfileHero = this.renderProfileHero.bind(this);
     this.renderPageBody = this.renderPageBody.bind(this);
     this.renderProfileView = this.renderProfileView.bind(this);
   }
 
-  componentWillMount() {
+  componentDidMount() {
     const { getUser, userId } = this.props;
     getUser(userId);
+    // eslint-disable-next-line react/no-did-mount-set-state
```
Comment (codrex) [inline src/containers/ProfileContainer/index.js:30]: // eslint-disable-next-line react/no-did-mount-set-state ??

Calling setState in `componentDidMount` causes the component to re-render. The better approach for this is to set `userFetched` to true in the constructor.

File: src/containers/ProfileContainer/index.js [hunk 2]
```diff
@@ -97,20 +107,58 @@ class ProfileContainer extends React.Component {
...
-  fetchIsLoading: PropTypes.bool
+  updateProfile: PropTypes.func.isRequired,
+  updateIsLoading: PropTypes.bool.isRequired,
+  fetchIsLoading: PropTypes.bool.isRequired
+};
+
+const mapDispatchToProps = dispatch => ({
+  getUser: userId => dispatch(fetchUser(userId)),
+  updateProfile: (payload, create) => dispatch(updateUserProfile(payload, create))
+});
```
Comment (codrex) [inline src/containers/ProfileContainer/index.js]: Connecting your action creator to props this way works, but a better approach is to make use of the bindActionCreators function that comes with the react-redux lib.

Comment (eltNEG) [reply src/containers/ProfileContainer/index.js]: I have implemented this. Thanks

File: src/containers/ProfileContainer/index.js (hunk 1 above)
Comment (eltNEG) [reply src/containers/ProfileContainer/index.js:30]: I have moved the `setState` to `componentWillRecieveProps`. Setting `userFetched` in in the constructor is too early for what I wanted to achieve.

Comment (codrex) [reply src/containers/ProfileContainer/index.js:30]: Then it is better off in componentDidMount since componentWillRecieveProps is a deprecated API

Comment (eltNEG) [reply src/containers/ProfileContainer/index.js:30]: I have moved it back to `componentDidMount`

**Review by codrex — APPROVED** [review]

**Pull request merged**